tests/TestPscPublish.hs

```diff
@@ -73,3 +65,15 @@ testPackage dir = pushd dir $ do
         putStrLn ("psc-publish tests failed on " ++ dir ++ ":")
         putStrLn (show other)
         exitFailure
+
+-- | Fail with a helpful error message when preparePackage fails
+preparePackageError :: Publish.PackageError -> IO ()
+preparePackageError e = do
+  Publish.printError e
```

Comment:
Is it possible to inspect this error to make a better guess at whether it's appropriate to print the submodules message?